shark_turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -0,0 +1,68 @@
+from ..ops.wave_ops import Write, ExtractSlice, get_custom
+from .constraints import Constraint, HardwareConstraint
+from .._support.tracing import CapturedTrace, IndexingContext
+from .._support.indexing import IndexSymbol, IndexSequence
+from ..lang.global_symbols import *
+from .utils import simplify_index, get_mma_dimensional_mapping, get_hardware_vector_size
+import torch.fx as fx
+import numpy as np
+
+
+def get_vector_shape(
+    trace: CapturedTrace,
+    hardware_constraint: HardwareConstraint,
+    symbolic_shape: list[IndexSymbol],
+) -> list[int]:
+    mma_indices = get_mma_dimensional_mapping(trace)
+    return [
+        get_hardware_vector_size(dim, hardware_constraint, mma_indices)
+        for dim in symbolic_shape
+    ]
+
+
+def partition_strided_operators(trace: CapturedTrace, constraints: list[Constraint]):
+    """
+    This function analyzes the index sequence of operators in the graph
+    that are writes on 2d tensors. If the operator has an access pattern where
+    the strides are greater than one on a single dimension, this function splits the
+    operands into individual elements and constructs a write for
+    each individual element.
+    """
+
+    def has_strided_access(node: fx.Node) -> bool:
+        """
+        Checks for writes on 2d tensors with strided access on a single dimension.
+        """
+        custom = get_custom(node)
+        if isinstance(custom, Write) and len(custom.type.symbolic_shape) == 2:
+            strides = [simplify_index(custom.index[dim]).stride for dim in custom.index]
+            return sum(1 for stride in strides if stride > 1) == 1
```

Comment:
Can we use `any(stride > 1 for stride in strides)` ?

Reply:
I think we'd need to add a Not Implemented error for cases where multiple dims are strided since we do not handle it yet.

Reply:
just added this in my next PR.